AppBuilder/platform/views/ABViewForm.js

```diff
@@ -1043,10 +1053,19 @@ module.exports = class ABViewForm extends ABViewFormCore {
 
             if (relationFieldCom == null) return;
 
-            var relationFieldView = this.viewComponents[relationFieldCom.id];
+            var relationFieldView = this.viewComponents[relationFieldCom.id].ui
+               .id;
+            if (
+               this.viewComponents[relationFieldCom.id].ui.rows &&
+               this.viewComponents[relationFieldCom.id].ui.rows[0] &&
+               this.viewComponents[relationFieldCom.id].ui.rows[0].id
+            ) {
+               var relationFieldView = this.viewComponents[relationFieldCom.id]
```

Comment:
This part of the code I refactored yesterday so I don't have to look so deep into this new ui structure. I will replace this with the easier to read code now.

Reply:
Now I store the UI id of the nested form input at the top level so you don't have to go digging for it. I needed the nested UI because the add new button is webix code now not just custom HTML as well as the rest of the component being webix code and not just selectivity HTML
![Uploading Screen Shot 2022-03-10 at 10.48.36 AM.png…]()

Reply:
okay got this refactored...lots of code that traversed the ui JSON is gone now.